hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -243,6 +245,24 @@ def check_valid_minmax(prefix, val, info_obj):
                 check_valid_interval(min_value, max_value, "min_value", "max_value")
             kw["max_value"] = max_value
 
+        # We infer whether an array module will flush subnormals to zero, as may
+        # be the case when libraries are built with compiler options that
+        # violate IEEE-754 (e.g. -ffast-math and -ftz=true). Note we do this for
+        # the specific dtype, as compilers may end up flushing subnormals for
+        # one float but supporting subnormals for the other.
+        #
+        # By default, floats() will generate subnormals if they are in the
+        # inferred values range. If we have detected that xp flushes to zero for
+        # the passed dtype, we ensure from_dtype() will not generate subnormals
+        # by default.
+        if allow_subnormal is not None:
+            kw["allow_subnormal"] = allow_subnormal
+        else:
+            subnormal = next_down(float(finfo.smallest_normal), width=finfo.bits)
```

Comment:
I'll check with NumPy folks and maybe raise an issue, as the mocked `finfo` type relies on a seemingly erroneous `np.finfo()`. I should maybe fix our mocking and remove this `float()` casting. `xp.finfo()` returns the Python `float` fortunately.